src/pulsar_producer.erl

```diff
@@ -702,7 +706,21 @@ enqueue_send_requests(Requests, State = #{replayq := Q}) ->
                Requests),
     NewQ = replayq:append(Q, QItems),
     ?tp(pulsar_producer_send_requests_enqueued, #{requests => Requests}),
-    State#{replayq := NewQ}.
+    Overflow = replayq:overflow(NewQ),
+    handle_overflow(State#{replayq := NewQ}, Overflow).
+
+-spec handle_overflow(state(), integer()) -> state().
+handle_overflow(State, Overflow) when Overflow =< 0 ->
+    %% no overflow
+    State;
+handle_overflow(State0 = #{replayq := Q, callback := Callback}, Overflow) ->
+    {NewQ, QAckRef, Items0} =
+        replayq:pop(Q, #{bytes_limit => Overflow, count_limit => 999999999}),
+    ok = replayq:ack(NewQ, QAckRef),
+    log_warn("replayq dropped ~b overflowed messages", [length(Items0)], State0),
```

Comment:
```suggestion
    log_warn("replayq dropped ~b overflown messages", [length(Items0)], State0),
```

Reply:
"Overflown" would be the past participle of "overfly", it seems.